pkg/controller/node/nodecontroller_test.go

```diff
@@ -1098,6 +1099,67 @@ func TestCloudProviderNoRateLimit(t *testing.T) {
 	}
 }
 
+func TestNodeEventGeneration(t *testing.T) {
```

Comment:
nit: can you please move the test where it was before? It would be easier for review :)